charts/consul/templates/mesh-gateway-deployment.yaml

```diff
@@ -86,160 +86,180 @@ spec:
-            - name: consul-ca-cert
-              mountPath: /consul/tls/ca
-              readOnly: true
-            {{- end }}
-          {{- if .Values.meshGateway.initServiceInitContainer.resources }}
-          resources: {{ toYaml .Values.meshGateway.initServiceInitContainer.resources | nindent 12 }}
+        - name: CONSUL_LOGIN_DATACENTER
+          {{- if and .Values.global.federation.enabled .Values.global.federation.primaryDatacenter }}
+          value: {{ .Values.global.federation.primaryDatacenter }}
+          {{- else }}
+          value: {{ .Values.global.datacenter }}
           {{- end }}
+        - name: CONSUL_LOGIN_META
+          value: "component=mesh-gateway,pod=$(NAMESPACE)/$(POD_NAME)"
+        {{- end }}
+        command:
+        - "/bin/sh"
+        - "-ec"
+        - |
+          consul-k8s-control-plane connect-init -pod-name=${POD_NAME} -pod-namespace=${NAMESPACE} \
+           -gateway-kind="mesh-gateway" \
+           -consul-node-name="k8s-service-mesh" \
+           -proxy-id-file=/consul/service/proxy-id \
+           -service-name={{ .Values.meshGateway.consulServiceName }} \
+           -log-level={{ default .Values.global.logLevel }} \
+           -log-json={{ .Values.global.logJSON }}
+        volumeMounts:
+        - name: consul-service
+          mountPath: /consul/service
+        {{- if .Values.global.tls.enabled }}
+        - name: consul-ca-cert
+          mountPath: /consul/tls/ca
+          readOnly: true
+        {{- end }}
+        {{- if .Values.meshGateway.initServiceInitContainer.resources }}
+        resources: {{ toYaml .Values.meshGateway.initServiceInitContainer.resources | nindent 10 }}
+        {{- end }}
       containers:
-        - name: mesh-gateway
-          image: {{ .Values.global.imageConsulDataplane | quote }}
-          {{- if .Values.meshGateway.resources }}
-          resources:
+      - name: mesh-gateway
+        image: {{ .Values.global.imageConsulDataplane | quote }}
+        {{- if .Values.meshGateway.resources }}
+        resources:
             {{- if eq (typeOf .Values.meshGateway.resources) "string" }}
             {{ tpl .Values.meshGateway.resources . | nindent 12 | trim }}
             {{- else }}
             {{- toYaml .Values.meshGateway.resources | nindent 12 }}
             {{- end }}
+        {{- end }}
+        volumeMounts:
+        - mountPath: /consul/service
+          name: consul-service
+          readOnly: true
+        {{- if .Values.global.tls.enabled }}
+        - name: consul-ca-cert
+          mountPath: /consul/tls/ca
+          readOnly: true
+        {{- end }}
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        command:
+        - /bin/sh
+        - -ec
+        - |
+          consul-dataplane \
```

Comment:
🤔 I'm not sure I'm following your line of thinking. Are you saying surround `consul-dataplane` in single quotes?

A user is already completely in charge of how this gets installed so I'm not sure why they would be malicious to their own installation. They can already do `helm template` and modify this deployment however they want. Let me know if I'm missing something though!

Reply:
Not the entire command, but values passed to the flags. You're right to assume that once the user has access to update the Helm installation, they already have a lot of permissions. I'm just wondering if there are any tools here for malicious data access that we can head off by single-quoting the values, (e.g. `-tls-server-name='{{.Values.externalServers.tlsServerName }}'`). Up to you.

Reply:
Since this would only be possible during templating, I'll leave it as is for now because like I mentioned before a user already has full control over this deployment and they can modify it to that effect outside of helm.